db/background_mgr_resync_dcp.go

```diff
@@ -209,6 +209,20 @@ func (r *ResyncManagerDCP) Run(ctx context.Context, options map[string]interface
 			}
 
 		}
+
+		// If we regenerated sequences, update syncInfo for all collections affected
+		if regenerateSequences {
```

Comment:
I think the latter approach (query resync) is slightly more preferable but also not possible to do with DCP resync, as it streams over multiple collections.

If we change query resync to match the DCP behaviour, you'd end up potentially re-running resync unnecessarily for a collection, if you stopped the resync process prematurely and it had only completed a subset of collections.

DCP doesn't have this advantage because the stream is ordered differently and all collections would resync the same rate (assuming even distribution of docs across collections)